core/src/main/java/io/questdb/cairo/mv/MatViewRefreshJob.java

```diff
@@ -434,27 +418,29 @@ private RefreshContext findRefreshIntervals(
             // For small sampling intervals such as '10T' or '2s' the actual sampler is
             // chosen as a 10x multiple of the original interval. That's to speed up
             // iteration done by the interval iterator.
-            final long minIntervalMicros = configuration.getMatViewMinRefreshInterval();
+            final long minInterval = driver.from(configuration.getMatViewMinRefreshInterval(), ColumnType.TIMESTAMP_MICRO);
             long approxBucketSize = timestampSampler.getApproxBucketSize();
             long actualInterval = viewDefinition.getSamplingInterval();
-            if (approxBucketSize < minIntervalMicros) {
-                while (approxBucketSize < minIntervalMicros) {
+            if (approxBucketSize < minInterval) {
+                while (approxBucketSize < minInterval) {
                     approxBucketSize *= 10;
                     actualInterval *= 10;
                 }
                 timestampSampler = TimestampSamplerFactory.getInstance(
+                        driver,
                         actualInterval,
                         viewDefinition.getSamplingIntervalUnit(),
                         0
                 );
             }
 
-            final long rowsPerBucket = estimateRowsPerBucket(baseTableReader, timestampSampler.getApproxBucketSize());
+            final long rowsPerBucket = estimateRowsPerBucket(driver, baseTableReader, timestampSampler.getApproxBucketSize());
             final int rowsPerQuery = configuration.getMatViewRowsPerQueryEstimate();
             final int step = Math.max(1, (int) (rowsPerQuery / rowsPerBucket));
 
             // there are no concurrent accesses to the sampler at this point as we've locked the state
             final SampleByIntervalIterator intervalIterator = intervalIterator(
```

Comment:
same story here, driver, sampler and rules have to align - we should definitely validate this is the case before throwing this at the runtime